File: sudoku.go
package sudoku

func IsValid(sudoku string) bool {
//BEGIN_CHALLENGE
	// Appereantly Go is quite strict about types and does not allow a int8 to be added an int
	var data [9][9]int; 
	sum := 0

	// Create array and compute field sum
	for x := 0; x < 9; x++ {
		for y := 0; y < 9; y++ {
			data[x][y] = int(sudoku[x * 9 + y] - 0x30)
			sum += data[x][y]
		}
	}

	// Check field sum
	if sum != 405 {
		return false
	}

	// Check cells
	for val := 1; val <= 9; val++ {
		for cell := 0; cell < 9; cell++ {
			count := 0
			sx := cell % 3
			sy := cell - sx
			sx *= 3

			for x := 0; x < 3; x++ {
				for y := 0; y < 3; y++ {
					if data[x + sx][y + sy] == val {
						count++
						if count > 1 {
							return false
						}
					}
				}
			}
		}
	}

	// Check rows
	for val := 1; val <= 9; val++ {
		for y := 0; y < 9; y++ {
			count := 0
			for x := 0; x < 9; x++ {
				if data[x][y] == val {
					count++
					if count > 1 {
						return false
					}
				}
			}
		}
	}

	// Check columns
	for val := 1; val <= 9; val++ {
		for x := 0; x < 9; x++ {
			count := 0
			for y := 0; y < 9; y++ {
				if data[x][y] == val {
					count++
					if count > 1 {
						return false
					}
				}
			}
		}
	}

//END_CHALLENGE
	return true
}

File: sudoku.py
def is_valid(sudoku):
    #BEGIN_CHALLENGE
    chars = list(sudoku)
    data = [[0 for x in range(9)] for x in range(9)]
    sum = 0
    for x in range(9):
        for y in range(9):
            data[x][y] = ord(chars[x + (y * 9)]) - 0x30
            sum += data[x][y]
    # check sum
    if sum != 405:
        return False

    # check cells
    for val in range(1, 10):
        for cell in range(9):
            count = 0
            sx = cell % 3
            sy = cell - sx
            sx *= 3

            for x in range(3):
                for y in range(3):
                    if data[x + sx][y + sy] == val:
                        count += 1
                        if count > 1:
                            return False

    # check columns
    for val in range(1, 10):
        for x in range(9):
            count = 0
            for y in range(9):
                if data[x][y] == val:
                    count += 1
                    if count > 1:
                        return False

    # check rows
    for val in range(1, 10):
        for y in range(9):
            count = 0
            for x in range(9):
                if data[x][y] == val:
                    count += 1
                    if count > 1:
                        return False

    #END_CHALLENGE
    return True